src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageFilterFactoryProvider.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+{
+    public static class PageFilterFactoryProvider
```

Comment:
Is this a copy of the FilterCache code?

I'm ok for now if you want to move forward with this PR and de-dupe in a follow up. My expectation is that we'll need want to de-dupe some of this, nothing about filters is really specific to pages.